iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractShuntCompensatorModel.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.impl;
+
+import com.powsybl.iidm.network.ShuntCompensatorModel;
+import com.powsybl.iidm.network.Validable;
+import com.powsybl.iidm.network.ValidationException;
+
+import java.util.Objects;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+abstract class AbstractShuntCompensatorModel implements ShuntCompensatorModelHolder {
+
+    protected ShuntCompensatorImpl shuntCompensator;
+
+    @Override
+    public void setShuntCompensator(ShuntCompensatorImpl shuntCompensator) {
+        this.shuntCompensator = Objects.requireNonNull(shuntCompensator);
+    }
```

Comment:
I'm not sure about this: the model should be attached to the ShuntCompensator in the constructor, like it's done for tap changer

Reply:
It doesn't work as tapchangers are created when the transformer is already built, which is not the case for models (they are created at the same time since it does not make sense to create a shunt without model).

Reply:
I can check however that the shunt is only defined once.